.github/workflows/ci.yml

```diff
@@ -228,11 +228,15 @@ jobs:
           # gnupg2 is for signing the resulting binary.
           install: |
             apt-get update -q -y
-            apt-get install -q -y build-essential libz-dev git python3-pip gnupg2 curl
+            apt-get install -q -y zip unzip build-essential libz-dev git python3-pip gnupg2 curl
           run: |
             set -e
             export PATH="/utils:$PATH"
             git config --global --add safe.directory "$(pwd)"
+            ./mill -i ci.setShouldPublish
+            if [[ $SHOULD_PUBLISH == "true" ]]; then
+              .github/scripts/generate-os-packages.sh
+            fi
```

Comment:
I am pretty sure this env variable thing here will not work, but I am waiting which error CI gives me.

Reply:
Sorry for my confusing message. I didn't expect that `./mill -I ci.setShouldPublish` fail in this [way](https://github.com/VirtusLab/scala-cli/pull/1748/checks#step:5:721).

I think, your solution is better and it should work.

Reply:
@mkurz I think, we should include this PR into the next release. Can you able to return removed lines in `ci.yml` and then I think, we can merge it PR.

Reply:
@lwronski The PR does not work correctly. When running `.github/scripts/generate-os-packages.sh` it overrides the generated aarch64 binary with the "wrong" one meant for x86 systems (I tested this by downloading the binary, I could not run it).
The problem is: In the script the function `generate_sdk` calls the `launcher` function [here](https://github.com/VirtusLab/scala-cli/blob/d790aff03c7ebc69e383e2aa8bd2f6647b76d90e/.github/scripts/generate-os-packages.sh#L124) which runs `mill -i copyTo "cli.nativeImage" "scala" 1>&2` [here](https://github.com/VirtusLab/scala-cli/blob/d790aff03c7ebc69e383e2aa8bd2f6647b76d90e/.github/scripts/generate-os-packages.sh#L27) which again builds the binary and overrides the one generated by the virtual machine (at least this is what I think is happening).

What do you think is the best way to avoid this?

Reply:
Maybe we should back to your first idea and run `.github/scripts/generate-os-packages.sh` inside the Docker container. In this case, it should properly detect the architecture and take the `aarch64` launcher in the `.github/scripts/generate-os-packages.sh` script.

Reply:
Yeah, I will try that again.

Reply:
The `grep -q...` returns 0 if it finds the searched string and enters the if branch.
Thats part of the github_env var simulation...